lib/Dialect/Torch/IR/TorchOps.cpp

```diff
@@ -3664,7 +3664,9 @@ OpFoldResult AtenSliceTensorOp::fold(FoldAdaptor adaptor) {
     return DenseElementsAttr::get(outType.toBuiltinTensor(), values);
   }
 
-  // If the input and output shapes are the same we can just fold:
+  // If the input and output shapes are the same & step == 1 we can fold:
+  if (step.getValue().getSExtValue() != 1)
```

Comment:
Ah, I think the issue is resolved with:

```suggestion
  if (!step || step.getValue().getSExtValue() != 1)
```

It was crashing because step failed the dyn cast on initial applications of the folder (this was happening on `ScaledDotProductAttentionDifferentModule_basic`).

Sorry for submitting the sloppy fix earlier!

Reply:
Thanks for debugging the issue! Commited the fix.